### ISSUE TITLE
Is the code right?

### ISSUE THREAD

**Issue by evanli05** [ShayanPersonal/stacked-autoencoder-pytorch#1]
Hi, I'm getting the classification accuracy of 0% on CIFAR 10 dataset using your code. It doesn't look right. Could you please help me out?

**Comment by 17764591637** [comment]
The code is correct and can be run. But I found that when running the CIFAR 10 dataset, classification accuracy only had a 33% correct rate.

**Comment by ShayanPersonal** [comment]
The 0% issue is to to outdated formatting in the print statement, I'll write a quick fix for that.

I also see the accuracy of the linear classifier trained on the autoencoder features comes out to about ~35%. This is a small toy model and the linear classifier isn't expected to get good results on CIFAR10. Try messing around with hyperparameters, using more/less layers, adding more/less noise, etc for better results. One explanation is that there isn't enough noise in the autoencoders and the hidden states it's learning are too trivial (because it's learning something close to the identity mapping).
